experimental/segmenter/src/lstm_provider.rs

```diff
@@ -0,0 +1,69 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Data provider struct defines for ICU4X component.
+
+use icu_provider::prelude::*;
+use icu_segmenter_lstm::structs;
+
+pub mod key {
+    use icu_provider::ResourceKey;
+
+    #[allow(dead_code)]
+    pub const SEGMENTER_LSTM_V1: ResourceKey =
+        icu_provider::resource_key!(Core, "segmenter_lstm", 1);
+}
+
+/// Marker type for LstmData.
+pub struct LstmDataV1Marker;
+
+impl<'data> DataMarker<'data> for LstmDataV1Marker {
+    type Yokeable = structs::LstmData;
+    type Cart = structs::LstmData;
+}
```

Comment:
Nit: This boilerplate is auto-generated from `#[icu_provider::data_struct]` and you should use that annotation instead of writing out the boilerplate.